src/PKSim.Presentation/Presenters/Simulations/SimulationCompoundEnzymaticProcessPresenter.cs

```diff
@@ -94,6 +99,39 @@ public IEnumerable<string> AllMetabolitesFor(SimulationEnzymaticProcessSelection
          }
       }
 
+      public void AddPartialProcessMappingBaseOn(SimulationEnzymaticProcessSelectionDTO processSelectionDTO)
+      {
+         var currentSimulationPartialProcess = processSelectionDTO.SimulationPartialProcess;
+         var newProcess = new SimulationPartialProcess
+         {
+            CompoundProcess = _notSelectedPartialProcess,
+            IndividualMolecule = currentSimulationPartialProcess.IndividualMolecule,
+         };
+
+         var newProcessDTO = mapFrom(newProcess);
+         var insertIndex = _allPartialProcessesDTO.IndexOf(processSelectionDTO);
+         _allPartialProcessesDTO.Insert(insertIndex + 1, newProcessDTO);
+
+         rebindPartialProcess();
+      }
+
+      public void DeletePartialProcessMapping(SimulationEnzymaticProcessSelectionDTO dto)
+      {
+         _allPartialProcessesDTO.Remove(dto);
+         rebindPartialProcess();
+      }
+
+      public bool CanDeletePartialProcess(SimulationEnzymaticProcessSelectionDTO dto)
+      {
+         return _allPartialProcessesDTO.Count(x => x.IndividualMolecule == dto.IndividualMolecule) > 1;
```

Comment:
We need to ensure we have at least one. Otherwise the user won't be able to add it back!